src/components/manifold-oauth/manifold-oauth.tsx

```diff
@@ -1,31 +1,71 @@
 import { Event, EventEmitter, Component, h, Prop } from "@stencil/core";
 import { AuthToken, PumaAuthToken } from "../../interface";
 
+declare global {
+  interface Window {
+    DD_LOGS: any;
+  }
+}
+
 @Component({
   tag: "manifold-oauth"
 })
 export class ManifoldOauth {
   @Prop() oauthUrl?: string = "https://login.manifold.co/signin/oauth/web";
   @Event() receiveManifoldToken: EventEmitter<AuthToken>;
 
+  private ddScript?: HTMLScriptElement;
+  private loadTime?: Date;
+
   tokenListener = (ev: MessageEvent) => {
     const pumaToken = ev.data as PumaAuthToken;
     this.receiveManifoldToken.emit({
       token: pumaToken.access_token,
       expiry: pumaToken.expiry,
       error: pumaToken.error
     });
+    if (window.DD_LOGS) {
+      window.DD_LOGS.logger.info("Token received", {
+        source: "shadowcat",
+        type: "token_received",
+        duration: new Date().getTime() - this.loadTime.getTime()
+      });
+    }
+  };
+
+  ddLoadListener = () => {
+    window.DD_LOGS.init({
+      clientToken: "<@DATADOG_CLIENT_KEY@>",
```

Comment:
Whoa! This must be the `rollup-plugin-replace`, right? I searched quite a while ago for a way to use environment variables in Stencil and I never found anything good. Is this the recommended method? Where did you find this?

Reply:
Found an article on medium where someone did this - https://medium.com/stencil-tricks/hide-environment-variables-in-your-stenciljs-project-b327cda6ad3e

Honestly I'm not sure why there's not a Rollup equivalent to `inline-environment-variables-webpack-plugin`, but 🤷‍♂️